snippets/macros.hpp

```diff
@@ -265,3 +266,20 @@ namespace VULKAN_HPP_NAMESPACE
 #  define VULKAN_HPP_DEFAULT_ARGUMENT_NULLPTR_ASSIGNMENT = nullptr
 #  define VULKAN_HPP_DEFAULT_DISPATCHER_ASSIGNMENT       = VULKAN_HPP_DEFAULT_DISPATCHER
 #endif
+
+#if !defined( VULKAN_HPP_EXPECTED ) && ( 23 <= VULKAN_HPP_CPP_VERSION ) && ( __has_include( <expected>))
```

Comment:
I would use the feature test macro `__cpp_lib_expected`. Clang has `<expected>` via GCC, but cannot use the feature due to missing full support for concepts.